pkg/api/kptfile/v1alpha2/validation.go

```diff
@@ -74,27 +73,12 @@ func (f *Function) validate(fnType string, idx int) error {
 				Reason: err.Error(),
 			}
 		}
-		configFields = append(configFields, "configPath")
-	}
-	if len(f.ConfigMap) != 0 {
-		configFields = append(configFields, "configMap")
-	}
-	if !IsNodeZero(&f.Config) {
-		config := yaml.NewRNode(&f.Config)
-		if _, err := config.GetMeta(); err != nil {
+		if len(f.ConfigMap) != 0 {
```

Comment:
I think having an explicit check for this condition outside this the `if` will be better because user may not get this error if they have an invalid `functionConfig` and also `configMap` at the same time, right ?

```
 if len(f.ConfigMap) != 0 && f.ConfigPath != "" {
   // throw error
 }
```

Reply:
done